src/input.rs

```diff
@@ -425,6 +434,84 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
         }
     }
 
+    /// Underline URLs and change the mouse cursor when URL hover state changes.
+    fn update_url_highlight(&mut self, point: Point, modifiers: ModifiersState) {
+        let mouse_mode =
+            TermMode::MOUSE_MOTION | TermMode::MOUSE_DRAG | TermMode::MOUSE_REPORT_CLICK;
+
+        // Only show URLs as launchable when all required modifiers are pressed
+        if self.mouse_config.url.modifiers.relaxed_eq(modifiers)
+            && (!self.ctx.terminal().mode().contains(TermMode::ALT_SCREEN) || modifiers.shift)
+        {
+            if let Some(Url { text, origin }) = self.ctx.terminal().url_search(point.into()) {
```

Comment:
This nesting and early `return` is pretty ugly but done for performance reasons. It would be nice to get the same effect with a little bit cleaner code.